sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLoggingPolicy.java

```diff
@@ -52,105 +57,142 @@ public HttpLoggingPolicy(HttpLogOptions httpLogOptions) {
      * @param prettyPrintJSON If true, pretty prints JSON message bodies when logging. If the detailLevel does not
      * include body logging, this flag does nothing.
      */
-    HttpLoggingPolicy(HttpLogOptions httpLogOptions, boolean prettyPrintJSON) {
-        this.httpLogOptions = httpLogOptions;
+    private HttpLoggingPolicy(HttpLogOptions httpLogOptions, boolean prettyPrintJSON) {
         this.prettyPrintJSON = prettyPrintJSON;
+
+        if (httpLogOptions == null) {
+            this.httpLogDetailLevel = HttpLogDetailLevel.NONE;
+            this.allowedHeaderNames = Collections.emptySet();
+            this.allowedQueryParameterNames = Collections.emptySet();
+        } else {
+            this.httpLogDetailLevel = httpLogOptions.getLogLevel();
+            this.allowedHeaderNames = httpLogOptions.getAllowedHeaderNames()
+                .stream()
+                .map(String::toLowerCase)
```

Comment:
Wouldn't this result in a spotbugs issue for not using `Locale.ROOT`?

Reply:
Apparently is doesn't, don't know if this is a method reference issue. Will update to use Locale.ROOT.